common/error_handler.go

```diff
@@ -13,10 +14,30 @@ const (
 	unknownError          = "Unknown Error"
 )
 
-func HandleError(ctx context.Context, err error) HTTPError {
+func HandleError(ctx context.Context, w http.ResponseWriter, kind Kind, message string, cause error, httpErrorMapper func(context.Context, error) *HTTPError) {
+	err := CreateError(ctx, kind, message, cause)
 	logEntry := GetLogEntryFromContext(ctx)
-	logEntry.Error(err)
```

Comment:
Can we just log the `ServerError` here instead of the `cause`? The `ServerError` contains the `cause`.